src/wallet/rpcwallet.cpp

```diff
@@ -3242,19 +3284,15 @@ UniValue z_getaddressforaccount(const UniValue& params, bool fHelp)
 
     std::set<libzcash::ReceiverType> receiverTypes;
     if (params.size() >= 2) {
-        const auto& parsed = params[1].get_array();
-        for (size_t i = 0; i < parsed.size(); i++) {
-            const std::string& p = parsed[i].get_str();
-            if (p == "p2pkh") {
-                receiverTypes.insert(ReceiverType::P2PKH);
-            } else if (p == "sapling") {
-                receiverTypes.insert(ReceiverType::Sapling);
-            } else if (p == "orchard") {
-                receiverTypes.insert(ReceiverType::Orchard);
-            } else {
-                throw JSONRPCError(RPC_INVALID_PARAMETER, "receiver type arguments must be \"p2pkh\", \"sapling\", or \"orchard\"");
-            }
-        }
+        receiverTypes = ReceiverTypesFromJSON(params[1].get_array())
+            .map_error([](const std::set<std::string>& invalidReceivers) {
+                throw JSONRPCError(
+                        RPC_INVALID_PARAMETER,
+                        strprintf(
+                            "\"%s\" is an invalid receiver type. Arguments must be \"p2pkh\", \"sapling\", or \"orchard\"",
+                            *invalidReceivers.begin()));
```

Comment:
```suggestion
                        strprintf(
                            "\"%s\" %s. Arguments must be \"p2pkh\", \"sapling\", or \"orchard\"",
                            invalidReceivers.size() > 1 ? "are invalid receiver types" : "is an invalid receiver type",
                            *invalidReceivers.begin()));
```

Reply:
Yeah, I can do this, but it needs a bigger change than this suggestion – I was just lazy and avoiding iterating over the set.

Reply:
I overkilled this for you.